src/rustzcash.rs

```diff
@@ -191,6 +191,44 @@ pub extern "system" fn librustzcash_merkle_hash(
     write_le(tmp, &mut result[..]);
 }
 
+
+/// Compute Sapling note commitment.
+#[no_mangle]
+pub extern "system" fn librustzcash_sapling_compute_cm(
+    diversifier: *const [c_uchar; 11],
+    pk_d: *const [c_uchar; 32],
+    value: uint64_t,
+    r: *const [c_uchar; 32],
+    result: *mut [c_uchar; 32],
+) -> bool {
+
+    let diversifier = sapling_crypto::primitives::Diversifier(unsafe { *diversifier });
+    let g_d = match diversifier.g_d::<Bls12>(&JUBJUB) {
+        Some(g_d) => g_d,
+        None => return false
+    };
```

Comment:
I was going to suggest using `sapling_crypto::primitives::PaymentAddress`, but that involves an unnecessary copy, so the current logic is fine.